### ISSUE TITLE
iperf2 test completes but pScheduler report failed run

### ISSUE THREAD

**Issue by tonin** [perfsonar/pscheduler#1050]
I'm seeing the following behavior:
```
geantadmin^2@psmall:)~$ pscheduler task --tool iperf2 throughput --dest perfsonar.eenet.ee
Submitting task...
Task URL:
https://localhost/pscheduler/tasks/de5f06c9-b656-4bc8-b31f-194e3d6069d5
Running with tool 'iperf2'
Fetching first run...

Next scheduled run:
https://localhost/pscheduler/tasks/de5f06c9-b656-4bc8-b31f-194e3d6069d5/runs/af1cc947-85c8-4ff8-8e51-e87ba12a3362
Starts 2020-08-06T12:14:59Z (~2 seconds)
Ends   2020-08-06T12:15:14Z (~14 seconds)
Waiting for result...

Run did not complete: Failed

Limit system diagnostics for this run:

  Hints:
    requester: ::1
    server: ::1
  Identified as everybody, local-interfaces
  Classified as default, friendlies
  Application: Hosts we trust to do everything
    Group 1: Limit 'always' passed
    Group 1: Want all, 1/1 passed, 0/1 failed: PASS
    Application PASSES
  Application: Defaults applied to non-friendly hosts
    Group 1: Limit 'innocuous-tests' failed: Passed but inverted
    Group 1: Limit 'throughput-default-time' passed
    Group 1: Limit 'idleex-default' failed: Test is not 'idleex'
    Group 1: Want any, 1/3 passed, 2/3 failed: PASS
    Application PASSES
  Proposal meets limits
  Priority set to default of 0


Diagnostics from localhost:
  /usr/bin/iperf -p 5001 -c perfsonar.eenet.ee -t 10 -m
  
  ------------------------------------------------------------
  Client connecting to perfsonar.eenet.ee, TCP port 5001
  TCP window size: 85.0 KByte (default)
  ------------------------------------------------------------
  [  3] local 161.53.30.17 port 53554 connected with 193.40.132.142 port 5001
  [ ID] Interval       Transfer     Bandwidth
  [  3]  0.0-10.1 sec  1007 MBytes   838 Mbits/sec
  [  3] MSS size 1448 bytes (MTU 1500 bytes, ethernet)
  

Error from localhost:
  No error.

Diagnostics from perfsonar.eenet.ee:
  No result was produced

Error from perfsonar.eenet.ee:
  No result was produced

No further runs scheduled.
```

On the initiating host, I have the diags output https://psmall.st.carnet.hr/pscheduler/tasks/de5f06c9-b656-4bc8-b31f-194e3d6069d5/runs/af1cc947-85c8-4ff8-8e51-e87ba12a3362 and on the destination host, I have a `process took too long to run` error: https://perfsonar.eenet.ee/pscheduler/tasks/2e3edc70-b911-464f-8856-c138776bb193/runs/1bfc7199-495d-4eac-bd70-8f036bda1552

**Comment by mfeit-internet2** [comment]
Likely bad duration calculation.  Fixed in 4.3.

**Comment by tonin** [comment]
Yes, as discussed and confirmed with testing, current status with iperf2 tests is that it fails when the remote end is a 4.2 machine.  We have the following table:
 - 4.2->4.2 fails
 - 4.2->4.3 works
 - 4.3->4.2 fails
 - 4.3->4.3 works

As this is fixed in the upcoming 4.3 version, I think this issue can be closed.
